Add Sound Manager singleton prefab & audio mixer

Fodderbump opened the pull request:
The Audio Mixer has Music and SFX > Footsteps under the Main group.

The Sound manager singleton implemented as a prefab with one child, which has an Audio Source component (Background Source) that is connected to the Music group. TJ's hide & seek BGM is attached and plays & plays on Start() as an example.

Fodderbump committed:
Add Sound Manager prefab & audio mixer

Sound manager currently implemented with one child (background music source that is connected to Music group & plays on start)

mmhousto commented on Assets/Scripts/Managers/SoundManagerScript.cs:
```diff
@@ -0,0 +1,42 @@
+        else
+        {
+            S = this;
+        }
```
Add DontDestroyOnLoad() to this, so it persist through out all scenes.

mmhousto commented on Assets/Scripts/Managers/SoundManagerScript.cs:
```diff
@@ -0,0 +1,42 @@
+
+    public AudioClip stealthMinigameBGM; // drag clips into this field
+    public AudioSource backgroundSource;
+
```
Add a reference for the AudioMixer and methods with float parameter to set the volume of each group.
`public AudioMixer masterMixer;`
Then to change volume you need to expose the parameters of the group and give a name. Then can call this for example to set the master volume.
`masterMixer.SetFloat("MasterVolume", newLevel)`
where "MasterVolume" is the name of exposed parameter for group and newLevel is a float parameter on method.

mmhousto reviewed (commented):
Looks good! I think we just need it to persist through scenes and methods to set volume of each group in the audio mixer.

mmhousto commented on Assets/Scripts/Managers/SoundManagerScript.cs:
```diff
@@ -0,0 +1,42 @@
+
+public class SoundManagerScript : MonoBehaviour
+{
+    private static SoundManagerScript S; // this is the singleton!
```
Singleton is usually named Instance.

I wonder if we can store all audio clips with enums and arrays?

Should also add a sfxSource for all global sound effects/UI sound effects.

tjtimusk committed:
Imported Movement SFX

Fodderbump committed:
Add more functionality to SoundManager

Added DontDestroyOnLoad() to soundmanager singleton so that it persists through all scenes

Exposed pitch & volume for all groups and subgroups, added public method to access set float

Renamed singleton to Instance & created an enum and array for audioclips

tjtimusk committed:
Removed the Silly Space

Fodderbump committed:
Merge remote-tracking branch 'origin/TJ' into Clytie

mmhousto reviewed (approved):
Looks good! We will need to add these to the BGM: Title Music, King’s Court, Thieves Faction, Scavengers Faction, Magicians Faction

We will also need an array for playerSFX and ambience will need one plus an audio source.

PlayerSFX:
Footsteps: 10x each Stone, Dirt, Grass and Wood, Crouched and Running
Life Layers: Chainmail, Armour, ‘Pack’ and jostles for jumping/Landing

Ambience: Castle, Exterior, Caravan, Cave, Tower

I am going to merge to test and setup volume sliders.
